articles/precarious-data.md

```diff
@@ -1,51 +1,217 @@
 
-When you use an `http://` or `https://` link to point to a webpage, image, spreadsheet, dataset, tweet, etc, you're identifying content by its location.  The link is an identifier that points to a particular location on the web, which corresponds to a particular server, or set of servers, somewhere on the web.  **Whoever controls that location controls the content.** That's how HTTP works. It's _location-addressed_. Even if a thousand people have downloaded copies of a file, meaning that the content exists in a thousand locations, HTTP points to a single location and whoever controls that location decides what content is returned when people use that HTTP link.
-They also decide whether to return any content at all.
+* If the link points to a server that has been turned off or had an outage, the link is useless - it doesn't give me a way to find the same content elsewhere.
+* If the link points to a server that has been hacked, I might have no way to know that I'm getting bad content from that hacked location.
+* If you give me a link and someone changes the content at that location, it's hard (or impossible) for me to know that the content I retrieve from that location is different from the content you _thought_ you were pointing to.
+* I might already have a copy of the content, meaning I don't really need to retrieve it again, but the link doesn't give me a way to know that I already have it. I would have to download the content again and compare.
+* If someone on my local network has a copy of the content, the link doesn't let me get the content from that alternate local source. It only lets me get the content from the location identified by the link. This is especially painful if my internet connection is slow or if I don't have an internet connection at all.
 
 ## The Solution: Identify Information by its Fingerprint, not its Location
 
-The alternative is to identify content by its "fingerprint" rather than identifying it by its location. That way, when someone says "Look at the content with this fingerprint" you can get it from anyone who has a copy of the content. To do this, we identify content by its cryptographic fingerprint, known as a cryptographic hash.
+The alternative is to identify content by its "fingerprint" rather than identifying it by its location. That way, when someone says "Look at the content with this fingerprint" you can get it from anyone who has a copy of the content. To do this, we identify content by its cryptographic hash. When we identify content in this way, using the content's fingerprint instead of its location to identify it, this is called _content-addressing_.
+The cryptographic hash for a piece of content never changes, which means **content addressing guarantees that the links will always return the same content**, regardless of _where_ I retrieve the content from.  You can do it by using a _content-addressed_ protocol like IPFS instead of using a _location-addressed_ protocol like HTTP.
```

Comment:
Describe what a cryptographic hash is, briefly. "A crypographic hash is a string of letters and numbers, something which is determined by a computational function that reads the data for a file and spits out a number. It always gives back the same hash for the same data (this is what is known as a deterministic function). So the hash never changes, which means...